pkg/controller/build/controller.go

```diff
@@ -347,6 +379,15 @@ func (c *Controller) waitForOperationAsync(build *v1alpha1.Build, op builder.Ope
 	return nil
 }
 
+func newBuildTimeoutError(message string) *errors.StatusError {
+	return &errors.StatusError{metav1.Status{
+		Status:  metav1.StatusFailure,
+		Code:    http.StatusGatewayTimeout,
+		Reason:  metav1.StatusReasonTimeout,
+		Message: message,
+	}}
+}
+
```

Comment:
Are you using the http package  and error to indicate an timeout from an external server? Does the other status errors include an http error?

Reply:
Status error package includes client timeouts but in this case I wanted to throw custom timeout error with more detail about build.

Reply:
I'm not actually sure what benefit using a `StatusError` gives us, can we just return `fmt.Errorf("Build timed out after %s", timeout)`?

Reply:
Yeah I am good with that